keps/sig-storage/20181111-extend-datasource-field.md

```diff
@@ -123,6 +123,18 @@ The primary risk of this feature is requesting a PVC DataSource when using a CSI
 
 Due to the similarities between Clones and Snapshots, it is possible that some back ends may require queiscing in-use volumes before cloning.  This proposal suggests that initially, if a csi plugin is unable to safely perform the requested clone operation, it's the csi plugins responsibility to reject the request.  Going forward, when execution hooks are available (currently being proposed for consistent snapshots), that same mechanism should be made generally usable to apply to Clones as well.
 
+## Test Plan
+
+### Unit tests
+
+Unit tests are already in place and include:
+* Unit tests ensure that spec that includes PVC DataSource is interpretted correctly
+* When a PVC DataSource is specified in the spec the resultant PVC object includes the cooresponding DataSource entry
+
+### E2E tests
+
+Require addition of E2E tests using the clone feature of the CSI host provisioner
```

Comment:
Should we also add some more specific test cases we want to add? Like integration with other volume features like Raw Block, concurrent cloning operations on the same volume, stress/failure tests, etc.

Reply:
cloning while the source is being deleted